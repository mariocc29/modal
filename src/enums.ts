export enum MessageType {
  INFO = 'info',
  WARNING = 'warning',
  ALERT = 'alert',
  SUCCESS = 'success'
}