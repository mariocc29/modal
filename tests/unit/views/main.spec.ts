import { VueWrapper, shallowMount } from '@vue/test-utils';
import Vuex, { Commit, Store } from "vuex";
import { ComponentPublicInstance } from 'vue';

import { MessageType } from '@/enums';
import { RootState } from '../mocks/state.interface';
import MainView from '@/views/main.vue';


describe('MainView', () => {
  let store: Store<RootState>;
  let wrapper: VueWrapper<ComponentPublicInstance>;

  beforeEach(() => {
    store = new Vuex.Store<RootState>({
      state: {
        showModal: false,
        kindOf: MessageType.INFO
      },
      mutations: {
        TOGGLE_MODAL(state) {
          state.showModal = !state.showModal;
        },
        SET_KIND_OF_MODAL(state, kindOf: MessageType) {
          state.kindOf = kindOf
        }
      },
    });

    wrapper = shallowMount(MainView, {
      global: {
        plugins: [store]
      }
    });
  });

  it('should match with the snapshot', () => {
    expect( wrapper.html() ).toMatchSnapshot()
  })

  it('commits TOGGLE_MODAL with Information State (Info)', async () => {
    (store.commit as jest.MockedFunction<Commit>) = jest.fn();

    await wrapper.find('[kindOf="info"]').trigger('click');
    
    const vm = wrapper.vm as any;
    expect(vm.kindOf).toEqual(MessageType.INFO);
    expect(vm.showCheckboxOption).toBeTruthy();

    expect(store.commit).toHaveBeenCalledWith('TOGGLE_MODAL');
    expect(store.commit).toHaveBeenCalledWith('SET_KIND_OF_MODAL', MessageType.INFO);
  });

  it('commits TOGGLE_MODAL with Success State', async () => {
    (store.commit as jest.MockedFunction<Commit>) = jest.fn();

    await wrapper.find('[kindOf="success"]').trigger('click');
    
    const vm = wrapper.vm as any;
    expect(vm.kindOf).toEqual(MessageType.SUCCESS);
    expect(vm.showCheckboxOption).toBeFalsy();

    expect(store.commit).toHaveBeenCalledWith('TOGGLE_MODAL');
    expect(store.commit).toHaveBeenCalledWith('SET_KIND_OF_MODAL', MessageType.SUCCESS);
  });

  it('commits TOGGLE_MODAL with Warning State', async () => {
    (store.commit as jest.MockedFunction<Commit>) = jest.fn();

    await wrapper.find('[kindOf="warning"]').trigger('click');
    
    const vm = wrapper.vm as any;
    expect(vm.kindOf).toEqual(MessageType.WARNING);
    expect(vm.showCheckboxOption).toBeFalsy();

    expect(store.commit).toHaveBeenCalledWith('TOGGLE_MODAL');
    expect(store.commit).toHaveBeenCalledWith('SET_KIND_OF_MODAL', MessageType.WARNING);
  });

  it('commits TOGGLE_MODAL with Alert State', async () => {
    (store.commit as jest.MockedFunction<Commit>) = jest.fn();

    await wrapper.find('[kindOf="alert"]').trigger('click');
    
    const vm = wrapper.vm as any;
    expect(vm.kindOf).toEqual(MessageType.ALERT);
    expect(vm.showCheckboxOption).toBeFalsy();

    expect(store.commit).toHaveBeenCalledWith('TOGGLE_MODAL');
    expect(store.commit).toHaveBeenCalledWith('SET_KIND_OF_MODAL', MessageType.ALERT);
  });
})