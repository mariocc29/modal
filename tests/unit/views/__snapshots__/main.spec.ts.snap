// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView should match with the snapshot 1`] = `
<section>
  <article>
    <button-component-stub kindof="info"></button-component-stub>
    <button-component-stub kindof="success"></button-component-stub>
    <button-component-stub kindof="warning"></button-component-stub>
    <button-component-stub kindof="alert"></button-component-stub>
  </article>
</section>
<modal-component-stub title="Ceci est le titre d'une modale d'info" state="info" closeonoutsideclick="true"></modal-component-stub>
`;
